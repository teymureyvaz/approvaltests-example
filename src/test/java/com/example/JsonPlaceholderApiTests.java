package com.example;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

public class JsonPlaceholderApiTests {

    @Test
    public void testGetUserBody() {
        JsonPlaceholderApi api = new JsonPlaceholderApi("users/1");
        api.get().then().statusCode(200);
        Approvals.verify(api.getBody());
    }

    @Test
    public void testGetUserFullResponse() {
        JsonPlaceholderApi api = new JsonPlaceholderApi("users/1");
        api.get();
        Approvals.verify(api.getFullResponse(true));
    }
}
