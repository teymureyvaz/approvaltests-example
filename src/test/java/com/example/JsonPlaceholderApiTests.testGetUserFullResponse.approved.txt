#####DATE
Content-Type=application/json; charset=utf-8
Transfer-Encoding=chunked
Connection=keep-alive
Report-To={"group":"heroku-nel","max_age":3600,"endpoints":[{"url":"https://nel.heroku.com/reports?ts=1720209314&sid=e11707d5-02a7-43ef-b45e-2cf4d2036f7d&s=Js%2FPTFZnOxzG%2FV8qx9v2Samp%2Bj9pgEs3y4YwJDOB6xY%3D"}]}
Reporting-Endpoints=heroku-nel=https://nel.heroku.com/reports?ts=1720209314&sid=e11707d5-02a7-43ef-b45e-2cf4d2036f7d&s=Js%2FPTFZnOxzG%2FV8qx9v2Samp%2Bj9pgEs3y4YwJDOB6xY%3D
Nel={"report_to":"heroku-nel","max_age":3600,"success_fraction":0.005,"failure_fraction":0.05,"response_headers":["Via"]}
X-Powered-By=Express
X-Ratelimit-Limit=1000
X-Ratelimit-Remaining=996
X-Ratelimit-Reset=1720209325
Vary=Origin, Accept-Encoding
Access-Control-Allow-Credentials=true
Cache-Control=max-age=43200
Pragma=no-cache
Expires=-1
X-Content-Type-Options=nosniff
Etag=W/"1fd-+2Y3G3w049iSZtw5t1mzSnunngE"
Via=1.1 vegur
CF-Cache-Status=HIT
Age=27378
Server=cloudflare
#####CF-RAY
Content-Encoding=gzip
alt-svc=h3=":443"; ma=86400

{
    "id": 1,
    "name": "Leanne Graham",
    "username": "Bret",
    "email": "Sincere@april.biz",
    "address": {
        "street": "Kulas Light",
        "suite": "Apt. 556",
        "city": "Gwenborough",
        "zipcode": "92998-3874",
        "geo": {
            "lat": "-37.3159",
            "lng": "81.1496"
        }
    },
    "phone": "1-770-736-8031 x56442",
    "website": "hildegard.org",
    "company": {
        "name": "Romaguera-Crona",
        "catchPhrase": "Multi-layered client-server neural-net",
        "bs": "harness real-time e-markets"
    }
}